### ISSUE TITLE
$-variables expanded correctly?

### ISSUE THREAD

**Issue by muellerto** [gokcehan/lf#403]
I'm experimenting with self defined commands on Windows. I'm using the interactive command prompt after I pressed `:` for that. Then I type a command definition like this:

`:cmd Status !svn st $f`

This defines indeed a new command, I can map it to a key combination, I can call it, so far so good.

The problem is that this command has `$f` still in the executed command line. In my case `$f` is not expanded. So my svn says correctly

`svn: warning: W155007: 'E:\iip\Dev\$f' is not a working copy`

What can I do here?

r14, Windows 10 1909

**Comment by muellerto** [comment]
Update: this behavior is also the case when you run an interactive shell command after `!` like

`!svn update $f`

$f is not expanded in my case.

**Comment by gokcehan** [comment]
@muellerto On windows, variables use the syntax `%f%` , assuming you're using `cmd.exe`. Closing this issue. Feel free to leave further comments if any.

**Comment by muellerto** [comment]
So the explanation is: lf uses a variable notation **depending on the default shell** of the machine.

The reason for that is not clear. Thought it would be completely in the responsibility of the program which provides the variables to expand them again when they are used somewhere. No, you split this and give a part of the responsibility to a shell. Why? Look, on Windows I can start lf like every other console process without any shell, it runs then (unlike on Linux systems) in a new window automatically provided by the OS. In this case the user doesn't even know what shell will be used and when and for what. The command I defined is again based on an external console program (running in the same window). I never mentioned a shell, for nothing. Probably it's used implicitly but indeed I must explicitly use the notation of the invisible.

**Comment by gokcehan** [comment]
@muellerto It's not only the variables, but all the command syntax (e.g. conditionals, definitions) uses the default shell. We simply pass the string in between `$...\n` or `${{ ... }}` to the underlying shell without interpreting it in any way. In your examples, there are only variables. These are referred as environmental variables in the documentation. Why did you assume `$f` should work? There are cases we refer windows-specific variables with `%` in the documentation. For example default variables like editor is referred as `$EDITOR` for unix and `%EDITOR%` on windows. Does that not suggest you're supposed to use `%` on windows?

**Comment by muellerto** [comment]
I made a bunch of commands now and it seems to work as expected.

But indeed I couldn't figure this out based on the F1 help text. My reasons are as follows:

- The term `%f%` doesn't occure in this text, `$f` is the only used notation in all examples.

- `%EDITOR%` and others are mentioned, but this is something different, these are known _external_ environment variables used also by other applications and defined normally by an _external_ configuration dialog provided by Windows. I did never guess any relationship between `%EDITOR%` and `%f%`.

- I didn't expect the use of an "underlaying shell". I expected that lf starts a child process having a proper environment and proper (this means: fully expanded) parameters.
